tools/install.sh

```diff
@@ -0,0 +1,214 @@
+#!/bin/bash
+
+THIS_DIR=$( cd -- "$( dirname -- "${0}" )" &> /dev/null && pwd )
+P4C_DIR=$(readlink -f ${THIS_DIR}/..)
+
+# Get distribution info
+DISTRO=$(lsb_release -i | awk -F: '{print $2}' | xargs)
+VERSION=$(lsb_release -r | awk -F: '{print $2}' | xargs)
+
+# Define supported distributions and versions
+SUPPORTED_DISTROS=("Ubuntu" "Pop")
+SUPPORTED_UBUNTU_VERSIONS=("20.04" "22.04" "24.04")
+
+# Function to check if an element exists in an array
+contains() {
+    local element
+    for element in "${@:2}"; do
+        if [[ "$element" == "$1" ]]; then
+            return 0
+        fi
+    done
+    return 1
+}
+
+# Check distribution and version
+if contains "$DISTRO" "${SUPPORTED_DISTROS[@]}"; then
+    if [[ "$DISTRO" == "Ubuntu" ]]; then
+        if contains "$VERSION" "${SUPPORTED_UBUNTU_VERSIONS[@]}"; then
+            echo "Supported distribution: $DISTRO $VERSION"
+            exit 0
+        else
+            echo "Error: Unsupported Ubuntu version: $VERSION"
+            exit 1
+        fi
+    else
+        # Assume all versions of Pop!_OS and Linux Mint are supported
+        echo "Supported distribution: $DISTRO $VERSION"
+    fi
+else
+    echo "Error: Unsupported distribution: $DISTRO"
+    exit 1
+fi
+
+# Whether to install dependencies required to run PTF-ebpf tests
+: "${INSTALL_PTF_EBPF_DEPENDENCIES:=OFF}"
+# The build generator to use. Defaults to Make.
+: "${BUILD_GENERATOR:="ninja"}"
+# BMv2 is enable by default.
+: "${ENABLE_BMV2:=ON}"
+# eBPF is enabled by default.
+: "${ENABLE_EBPF:=ON}"
+# This is the list of back ends that can be enabled.
+# Back ends can be enabled from the command line with "ENABLE_[backend]=TRUE/FALSE"
+ENABLE_BACKENDS=("TOFINO" "BMV2" "EBPF" "UBPF" "DPDK"
+                 "P4TC" "P4FMT" "P4TEST" "P4C_GRAPHS"
+                 "TEST_TOOLS"
+)
+function build_cmake_enabled_backend_string() {
+  CMAKE_ENABLE_BACKENDS=""
+  for backend in "${ENABLE_BACKENDS[@]}";
+  do
+    enable_var=ENABLE_${backend}
+    if [ -n "${!enable_var}" ]; then
+      echo "${enable_var}=${!enable_var} is set."
+      CMAKE_ENABLE_BACKENDS+="-D${enable_var}=${!enable_var} "
+    fi
+  done
+}
+
+
+. /etc/lsb-release
+
+# In Docker builds, sudo is not available. So make it a noop.
+if [ "$IN_DOCKER" == "TRUE" ]; then
+  echo "Executing within docker container."
+  function sudo() { command "$@"; }
+fi
+
+
+# ! ------  BEGIN CORE -----------------------------------------------
+P4C_DEPS="bison \
+          flex \
+          cmake \
+          build-essential \
+          ccache \
+          ninja-build \
+          g++ \
+          git \
+          lld \
+          libboost-graph-dev \
+          libboost-iostreams-dev \
+          libfl-dev \
+          python3 \
+          python3-pip \
+          python3-setuptools"
+
+sudo apt-get update
+sudo apt-get install -y --no-install-recommends ${P4C_DEPS}
+
+sudo apt-get install -y python3-poetry python3-venv
+poetry install
+poetry shell
```

Comment:
Is my understanding correct that [uv](https://github.com/astral-sh/uv) is a poetry replacement, and if we use uv, there is no need to install poetry at all?

Reply:
Yeah, I am trying out `uv` as an alternative and it is pretty neat for dependency management so far.